src/vehicle.cpp

```diff
@@ -2467,9 +2469,7 @@ int vehicle::index_of_part(const vehicle_part *const part, bool const check_remo
       if (!check_removed && next_part.removed) {
         continue;
       }
-      if(part->get_id() == next_part.get_id() &&
-              part->mount == next_part.mount &&
-              part->hp == next_part.hp) {
+      if( part->get_id() == next_part.get_id() && part->mount == next_part.mount ) {
         return index;
       }
     }
```

Comment:
Agreed, but can that wait as further refactoring might allow us to drop that function entirely.